src/LoggyCore.hs

```diff
@@ -14,10 +14,14 @@ extractTimestamp dformat line = parsedTs
             _ -> error "Invalid log line!"
         timeParser = readSTime True defaultTimeLocale dformat  
 
-mergeLogLines :: DateFormat -> [LogLine] -> [LogLine] -> [LogLine]
-mergeLogLines dformat l1 l2 = sortedLogLines
+mergeLogLines :: DateFormat -> [[LogLine]] -> [LogLine]
+mergeLogLines dformat lstLines = mergeLogLines' dFormatPerFile lstLines
+    where dFormatPerFile = replicate (length lstLines) dformat
+
+mergeLogLines' :: [DateFormat] -> [[LogLine]] -> [LogLine]
```

Comment:
I would prefer this type sig:
```suggestion
mergeLogLines' :: [(DateFormat, [LogLine])] -> [LogLine]
```
to make it simpler to implement this fn, and to clarify what the relationship is between the format and lines

Reply:
Awesome suggestion, done.